### DISCUSSION TITLE
Umami v2.10.0 Cloud Not Capturing Referrer Data on Nuxt3 Static Site, Despite Correct Tracking in Vercel Analytics

### DISCUSSION THREAD

**Discussion by atom63** [umami-software/umami#2526] [Q&A]
Hello,

I am facing a specific issue with Umami Analytics (Cloud version v2.10.0) where referrer data fails to be tracked or displayed for my website. This site is built using Nuxt3, deployed as a static site on Vercel. Intriguingly, Vercel Analytics is capturing and displaying referrer data accurately, which indicates that the data is being sent correctly by the client. However, this is not reflected in my Umami dashboard, where all traffic sources are listed as (none).

**Environment:**
- Framework: Nuxt3 (Static Site Generation)
- Deployment: Vercel
- Analytics: Umami v2.10.0 Cloud

**Issue Description:**

While Vercel Analytics shows a comprehensive breakdown of referrers, Umami does not capture this data under the same conditions. This discrepancy suggests a specific issue with how Umami is handling or interpreting referrer data from my site.

![image](https://github.com/umami-software/umami/assets/25135269/ef44184c-c62e-4ab3-9773-498334eb9031)
![image](https://github.com/umami-software/umami/assets/25135269/70a16619-f137-4195-b8d0-e14dc2cf0768)

**Steps Taken:**
- Confirmed correct installation and loading of the Umami tracking script on all pages.
- Checked for restrictive Referrer-Policy settings or meta tags but found none that would impact tracking differently between Vercel and Umami.
- Ensured the absence of JavaScript errors that could interfere with tracking operations.
- Reviewed network requests to ensure referrer information is being sent but found Umami's captures to differ.

**Questions:**
- Are there known issues with Umami (specifically the Cloud version) in capturing or displaying referrer data for static sites on Nuxt3, especially when similar analytics services like Vercel Analytics do not exhibit these problems?
- Could a specific setting or configuration in Umami be required to ensure referrer data is captured and displayed correctly in this scenario?
- Is there a way to enable more detailed logging or debugging within Umami to trace how it processes incoming referrer data?

I am seeking advice on resolving this discrepancy or any recommended steps for further troubleshooting. Insights from the Umami community or the development team would be highly valuable and appreciated.

Thank you for your assistance and for providing a platform for analytics that respects user privacy.

**Comment by mikecao** [comment]
Vercel Analytics is not open-source, so we can't determine how they get their data. I assume they use a combination of client requests and logs that they capture on their servers. How Umami does it is simple. It just checks if there is a `document.referrer` value and uses that.

To debug, you can open your network panel in your browser developer tools to verify the payload. It should look something like this

<img width="497" alt="image" src="https://github.com/umami-software/umami/assets/477975/09888d68-51dc-45bc-9261-cb31b7d40ba5">

See the referrer that is sent.

**Comment by erkage** [comment]
Hey @mikecao,

I have the same issue but in a Vue SPA.
Here is a screenshot which shows that there might be some problem with the referrers:
<img width="574" alt="Screenshot 2024-04-02 at 22 05 14" src="https://github.com/umami-software/umami/assets/4376696/eb924d73-4009-4627-bcc7-758de7c88e52">

I also using [ackee](https://ackee.electerious.com/) which shows the referrers correctly and it seems it also using the [`document.referrer`](https://github.com/search?q=repo:electerious/ackee-tracker%20referrer&type=code).

**Reply by mikecao** [reply]
Is your SPA redirecting somewhere? The initial referrer always comes from `document.referrer`. But if something pushes to history, then the referrer changes to what that previous page was.

**Reply by erkage** [reply]
It was a direct page load from facebook to the main index page and there shouldn't be any redirection or manual history push in this case, unless VueRouter do something in the background but I don't think so.

**Reply by erkage** [reply]
However I found that when I just load the main page the `history.state` looks like this:
```
{
    "back": null,
    "current": "/",
    "forward": null,
    "position": 0,
    "replaced": true,
    "scroll": {
        "left": 0,
        "top": 0
    }
}
```
Where the `replaced` value is `true`, also as I see on a simple html page the state initially should be `null`.

I'll dig deeper into this.

**Reply by emsitkowski** [reply]
I have the same issue with Nuxt3 and Vercel. Found any solution?

**Reply by mikecao** [reply]
We've updated the tracker script. Can you check if you are still having issues?
